feature-libs/requested-delivery-date/root/components/shared/date-validation.service.ts

```diff
@@ -21,8 +21,37 @@ export class DateValidationService {
       value !== undefined &&
       value.length > 0 &&
       !isNaN(
-        new Date(value.replace(/(\d{2})-(\d{2})-(\d{4})/, '$2/$1/$3')).getDate() //convert 'dd-mm-yyyy' into 'mm/dd/yyyy'
+        this.getDateFromDateString(value).getDate() //convert 'dd-mm-yyyy' into 'mm/dd/yyyy'
       )
     );
   }
+
+  /**
+   * Compares 2 Date strings in the format 'dd-mm-yyy'
+   * @param date1
+   * @param date2
+   * @returns -1 if date2 is greater, 0 if both the dates are equal, 1 if date1 is greater, -2 for invalid inputs
+   */
+  compareDateStrings(date1: string, date2: string): number {
+    if (date1.length === 0 || date2.length === 0) {
```

Comment:
I think this function can be simplified, as right now there is a lot of if's.

```
compareDateStrings(date1: string, date2: string): number {
  if (!date1 || !date2) {
    return -2;
  }
  
  const d1 = this.getDateFromDateString(date1);
  const d2 = this.getDateFromDateString(date2);
  
  return d1 === d2 ? 0 : d1 < d2 ? -1 : 1;
}
```

Also could you let me know, why as a result for any specific case we need to return a number (like -2  for example) instead of some `enum` value with the status?

Reply:
returning a number would make it easier to compare by checking `compareDateStrings(date1, date2) < 0`

Reply:
Still I am not sure if I get a concept here. You are calling this `compareDateStrings` once in a component's statement:

```
 this.dateValidationService.compareDateStrings(
        requestedDate,
        this.earliestRetrievalAt || ''
      ) < 0
```

by `< 0` you expect date2 to be greater or -2 for invalid inputs. I guess we could change a logic of your function and return `true` or `false` depending on an arguments passed to the function. There is not need to do this check in a component in my opinion.

Reply:
The logic in the lines https://github.com/SAP/spartacus/blob/4f88f530773a5c4f5dffa17405487774767ac573/feature-libs/requested-delivery-date/root/components/delivery-mode-date-picker/delivery-mode-date-picker.component.ts#L109-L112 
check for invalid date inputs, so the `compareDateStrings` method will not be passed an invalid input.

That being said, we return `-2` as a safety in case this method gets reused in the future.

We can definitely write a wrapper method that returns a `boolean` like - 

```
isDateGreaterOrEqual(source: string, target: string): boolean {
    return this.compareDateStrings(source, target) >= 0;
  }
```

What do you suggest @Matejk00 ?

Reply:
If we are using `compareDateStrings` only in one place and it's not used anywhere else, we could transform this method to be `isDateGreaterOrEqual`. No need to wrap stuff in my opinion, just change a name and a logic of the existing method.

What do you think @SherwinVarghese?

Reply:
okay @Matejk00, sounds good.